static/js/services/enketo.js

```diff
@@ -452,7 +453,9 @@ angular.module('inboxServices').service('Enketo',
     var getUserContact = function() {
       return UserContact().then(function(contact) {
         if (!contact) {
-          throw new Error('Your user does not have an associated contact, or does not have access to the associated contact. Talk to your administrator to correct this.');
+          var err = new Error('Your user does not have an associated contact, or does not have access to the associated contact. Talk to your administrator to correct this.');
+          err.type = NO_ASSOCIATED_CONTACT_ERROR;
```

Comment:
I wonder instead of having a `type` set to a constant would it make more sense to have a `translationKey` property which you could set here. Then if it's not set it could be defaulted to whatever in the controller. That would remove some duplicated logic and an app-wide constant.

Reply:
I thought about that, however I was wondering if there are usecases when someone would need to use this logic (checking for an associated contact) but display a different error message. I guess not based on the message being used to create the `Error` now that I think about it. I'm happy to change it and then if that usecase does come up we can think about adding the `type` property back in or another solution?

Reply:
Yeah let's add it if/when it comes up.